cypress/integration/community.spec.js

```diff
@@ -0,0 +1,64 @@
+/// <reference types="cypress" />
+// Tests the functionality from the Community member perspective
+
+describe("Community members", () => {
+    beforeEach(() => {
+
+        cy.visit(Cypress.env('login_url'));
+        cy.get("[name=csrfmiddlewaretoken]")
+            .should("exist")
+            .should("have.attr", "value")
+            .as("csrfToken");
+
+        cy.get("@csrfToken").then((token) => {
+            cy.request({
+                method: "POST",
+                url: Cypress.env("login_url"),
+                form: true,
+                body: {
+                    username: "community",
+                    password: "1234567890",
+                },
+                headers: {
+                    "X-CSRFTOKEN": token,
+                },
+            });
+        });
+    })
+
+    it("can mark a translation as good or bad", () => {
+        cy.visit(Cypress.env('home'));
+
+        cy.get('[data-cy="segment-card"]:first')
+            .within(() => {
+
+                cy.get('[data-cy="yes-button"]')
+                    .should('be.visible')
+                    .click()
+                    .should('have.css', 'background-color', 'rgba(102, 142, 63, 0.5)')
```

Comment:
These hard-coded colour values are probably going to be a problem at some point.

Maybe just set the values as constants in a shared JS file, with a comment in both the CSS and JS file saying that if you update one you should update the other too?

Some other quick ideas, which may not be any better than the current thing:

  - Run some JavaScript in Cypress against the page under test to pull out variable definitions [[1]](https://stackoverflow.com/questions/36088655/accessing-a-css-custom-property-aka-css-variable-through-javascript), [[2]](https://github.com/cypress-io/cypress/issues/897)

        cy.window().then((win) => {
            const customGreen = win.getComputedStyle(win.document.documentElement).getPropertyValue('--custom-green');
            // returns " #668E3F"
        })

  - Add some non-displayed divs on the page to grab the CSS value from and
    compare against the target:

        <div
          data-cy="empty-div-with-custom-green-bg"
          style="background: var(--custom-purple); display: none;"
        />

Reply:
Testing for the correct colour is proving to be difficult. I agree that hard coding it is not the way to go. I can store the colour in a variable as suggested, but the variable returns a hex code and the test needs it to be in rgb. I'm wondering if testing for the colour change is even worth it here. What do you think?

Reply:
Yeah, testing for an exact colour is probably not worth it. Maybe just check for whether the correct css class is applied? Or if the test is really to make sure that something is reddish or greenish, [this stackoverflow answer](https://stackoverflow.com/a/38963890/14558) actually looks like it *might* be a useful way to map an arbitrary hex / RGB colour into one of eight major colours?

Reply:
I must have missed the `have.class` option when I was first writing this, 'cause that's exactly what I need. No more rgb checking!